api/types.proto

```diff
@@ -281,15 +281,32 @@ message UpdateConfig {
 	enum FailureAction {
 		PAUSE = 0;
 		CONTINUE = 1;
-		// TODO(aaronl): Add ROLLBACK as a supported failure mode.
-		// (#486)
+		ROLLBACK = 2;
 	}
 
 	// FailureAction is the action to take when an update failures.
-	// Currently, a failure is defined as a single updated task failing to
-	// reach the RUNNING state. In the future, there will be configuration
-	// to define what is treated as a failure (see #486 for a proposal).
 	FailureAction failure_action = 3;
+
+	// AllowedFailureFraction is the fraction of tasks that may fail during
+	// an update before the failure action is invoked. Any task created by
+	// the current update which ends up in one of the states REJECTED,
+	// COMPLETED or FAILED while the update is still running is counted as
+	// a failure. The number of failures is divided by the number of tasks
```

Comment:
That's a good point. What if we were to version specs? For example, we could keep a counter inside the Service object of how many times it has been updated, and copy that counter to the Task objects. By comparing this version number, we could distinguish tasks created by different updates, even if the specs match.